.github/workflows/release-create.yml

```diff
@@ -88,96 +76,154 @@ jobs:
-        grep -q " ${VERSION_PATTERH} " < site/docs/try/server-upgrade.md || FAILS="${FAILS} site/docs/try/server-upgrade.md"
-        grep -q "^| ${VERSION_PATTERH} .*check_mark.*$" < SECURITY.md || FAILS="${FAILS} SECURITY.md"
+        # version pattern without patch version
+        VERSION_PATTERN="$(echo ${RELEASE_VERSION} | sed 's/\([^.]*\)[.]\([^.]*\)[.].*/\1\\.\2/')\\."
+        grep -q " ${VERSION_PATTERN}" < site/docs/try/server-upgrade.md || FAILS="${FAILS} site/docs/try/server-upgrade.md"
+        grep -q "^| ${VERSION_PATTERN}" < README.md || FAILS="${FAILS} README.md"
+        grep -q "^| ${VERSION_PATTERN}.* .*check_mark.*$" < SECURITY.md || FAILS="${FAILS} SECURITY.md"
         if [[ -n ${FAILS} ]] ; then
-          echo ${FAILS} "do not contain the release version ${RELEASE_VERSION}."
+          echo ${FAILS} "do not match the version pattern ${VERSION_PATTERN}."
           exit 1
         fi
 
-    - name: Bump Python release version ${{ github.event.inputs.releaseVersion }}
-      working-directory: ./python
-      run: |
-        # bump2version 1.0.1 has a bug: https://github.com/c4urself/bump2version/issues/214
-        if [[ "$(cd python/ ; python -c 'import pynessie; print(pynessie.__version__)')" != ${RELEASE_VERSION} ]] ; then
-          bump2version --no-commit --no-tag --new-version ${RELEASE_VERSION} minor
-          # Call into pynessie to ensure bump2version didn't mess up anything
-          echo "pynessie at release-version $(python -c 'import pynessie; print(pynessie.__version__)')"
-        else
-          echo "pynessie already at release-version ${RELEASE_VERSION}"
-        fi
-
-    - name: Bump versions in site/ to ${{ github.event.inputs.releaseVersion }}
-      working-directory: ./site
-      run: |
-        cp mkdocs.yml /tmp/nessie-mkdocs.yml
-        cat /tmp/nessie-mkdocs.yml | sed "s/^    java: [0-9.]*$/    java: ${RELEASE_VERSION}/" | sed "s/^    python: v[0-9.]*$/    python: v${RELEASE_VERSION}/"> mkdocs.yml
+    - name: Bump versions for Python, site/, helm and UI
+      uses: ./.github/actions/bump-versions
+      with:
+        new-version: ${{ env.RELEASE_VERSION }}
 
-    - name: Bump version in helm/nessie to ${{ github.event.inputs.releaseVersion }}
-      working-directory: ./helm/nessie
+    # Updates the site/docs/try/releases.md file, takes the first four lines (the big heading),
+    # adds a heading with the version and a note referencing the release notes on GitHub,
+    # followed by the Git change log, finalized with the remainder of the
+    # previous site/docs/try/releases.md file.
+    - name: Update releases.md file
       run: |
-        cp Chart.yaml /tmp/nessie-Chart.yaml
-        cat /tmp/nessie-Chart.yaml | sed "s/^version: [0-9.]*$/version: ${RELEASE_VERSION}/" > Chart.yaml
-
-    - name: Bump UI release version ${{ github.event.inputs.releaseVersion }}
-      working-directory: ./ui
+        head -4 site/docs/try/releases.md > /tmp/releases.md
```

Comment:
"top 4 lines"-logic seems brittle but no immediate idea how to make this more universal :sweat_smile: